lib/class-wp-json-comments-controller.php

```diff
@@ -0,0 +1,476 @@
+		}
+
+		if ( $force ) {
+			return array( 'message' => __( 'Permanently deleted comment' ) );
+		}
+
+		return array( 'message' => __( 'Deleted comment' ) );
+	}
+
+	/**
+	 * Prepare a single comment output for response.
+	 *
+	 * @param  object          $comment Comment object.
+	 * @param  WP_JSON_Request $request Request object.
+	 * @return array $fields
+	 */
+	public function prepare_item_for_response( $comment, $request ) {
+		$fields = array(
+			'id'           => (int) $comment->comment_ID,
+			'post_id'      => (int) $comment->comment_post_ID,
+			'parent_id'    => (int) $comment->comment_parent,
+			'user_id'      => (int) $comment->user_id,
+			'author'       => $comment->comment_author,
+			'author_email' => $comment->comment_author_email,
+			'author_url'   => $comment->comment_author_url,
+			'date'         => json_mysql_to_rfc3339( $comment->comment_date ),
+			'content'      => array(
+				'rendered'     => apply_filters( 'comment_text', $comment->comment_content, $comment ),
+			),
+			'status'       => $this->prepare_status_response( $comment->comment_approved ),
+			'type'         => get_comment_type( $comment->comment_ID ),
+		);
+
+		if ( 'edit' == $request['context'] ) {
+			$fields['date_gmt']       = json_mysql_to_rfc3339( $comment->comment_date_gmt );
+			$fields['content']['raw'] = $comment->comment_content;
+		}
+
+		$links = array();
+
+		if ( 0 !== (int) $comment->user_id ) {
+			$links['author'] = array(
+				'href' => json_url( '/wp/users/' . $comment->user_id ),
+			);
+		}
+
+		if ( 0 !== (int) $comment->comment_post_ID ) {
+			$links['post'] = array(
+				'href' => json_url( '/wp/posts/' . $comment->comment_post_ID ),
+			);
+		}
+		
+		if ( 0 !== (int) $comment->comment_parent ) {
+			$links['in-reply-to'] = array(
+				'href' => json_url( sprintf( '/wp/comments/%d', (int) $comment->comment_parent ) ),
+			);
+		}
+
+		$fields['_links'] = $links;
+
+		return apply_filters( 'json_prepare_comment', $fields, $comment, $request );
+	}
+
+	/**
+	 * Filter query parameters for comments collection endpoint.
+	 *
+	 * Prepares arguments before passing them along to WP_Comment_Query.
+	 *
+	 * @param  WP_JSON_Request $request Request object.
+	 * @return array           $prepared_args
+	 */
+	protected function prepare_items_query( $request ) {
+		$prepared_args = array(
+			'number'  => absint( $request['per_page'] ),
+			'post_id' => isset( $request['post_id'] ) ? absint( $request['post_id'] ) : '',
+			'parent'  => isset( $request['parent_id'] ) ? intval( $request['parent_id'] ) : '',
+			'search'  => $request['search'] ? sanitize_text_field( $request['search'] ) : '',
+			'orderby' => sanitize_key( $request['orderby'] ),
+			'order'   => sanitize_key( $request['order'] ),
+			'status'  => 'approve',
+			'type'    => 'comment',
+		);
+
+		$prepared_args['offset'] = $prepared_args['number'] * ( absint( $request['page'] ) - 1 );
+
+		if ( current_user_can( 'edit_posts' ) ) {
```

Comment:
@rmccue @justinshreve @joehoyle  I expanded the available /comments query parameters BUT I limited the parameters available to unauthenticated users.  I tried to separate what is "public" vs "private" using some logic.  Any concerns here?